### PR TITLE
Fix: image onLoad props

### PR DESCRIPTION
**Pull request by bohdanprog**
# Summary
Change data structure returned from the Image onLoad event.
Add fix in new arch for returned sizes.

## Test Plan

Manually test in both architectures and platforms.

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |      ✅     |
| Android |    ✅    |

### CODE REVIEW

File: apple/Elements/RNSVGImage.mm [hunk 1]
```diff
@@ -133,9 +143,12 @@ - (void)didReceiveImage:(UIImage *)image metadata:(id)metadata fromObserver:(voi
     // See for more info: T46311063.
     return;
   }
-    auto imageSource = _state->getData().getImageSource();
-    imageSource.size = {image.size.width, image.size.height};
-    static_cast<const RNSVGImageEventEmitter &>(*_eventEmitter).onLoad({imageSource.size.width, imageSource.size.height, imageSource.uri});
+  auto imageSource = _state->getData().getImageSource();
+  imageSource.size = {image.size.width, image.size.height};
+    static_cast<const RNSVGImageEventEmitter &>(*_eventEmitter).onLoad({.source = {
+        .width = imageSource.size.width * imageSource.scale,
```
Comment (WoLewicki) [inline apple/Elements/RNSVGImage.mm:149]: Why did you change the logic here to also include the scale?

File: apple/Elements/RNSVGImage.mm [hunk 2]
```diff
@@ -133,9 +143,12 @@ - (void)didReceiveImage:(UIImage *)image metadata:(id)metadata fromObserver:(voi
     // See for more info: T46311063.
     return;
   }
-    auto imageSource = _state->getData().getImageSource();
-    imageSource.size = {image.size.width, image.size.height};
-    static_cast<const RNSVGImageEventEmitter &>(*_eventEmitter).onLoad({imageSource.size.width, imageSource.size.height, imageSource.uri});
+  auto imageSource = _state->getData().getImageSource();
+  imageSource.size = {image.size.width, image.size.height};
+    static_cast<const RNSVGImageEventEmitter &>(*_eventEmitter).onLoad({.source = {
```
Comment (WoLewicki) [inline apple/Elements/RNSVGImage.mm:148]: I think it is nice to add a check for if `_eventEmitter` is nil, same as in RNScreens: https://github.com/software-mansion/react-native-screens/blob/a9bc1118b755c362a981638c0ce0091affd46fe0/ios/RNSScreen.mm#L323.

File: apple/Elements/RNSVGImage.mm [hunk 3]
```diff
@@ -203,18 +216,16 @@ - (void)setSrc:(RCTImageSource *)src
...
-                          });
-                          [self invalidate];
-                        });
-                      }];
+                       dispatch_async(dispatch_get_main_queue(), ^{
+                         self->_image = CGImageRetain(image.CGImage);
+                         self->_imageSize = CGSizeMake(CGImageGetWidth(self->_image), CGImageGetHeight(self->_image));
+                         if (self->_onLoad) {
+                           RCTImageSource *sourceLoaded = [src imageSourceWithSize:image.size scale:image.scale];
+                           self->_onLoad(onLoadParamsForSource(sourceLoaded));
```
Comment (WoLewicki) [inline apple/Elements/RNSVGImage.mm:224]: Does it make sense to add it if we only use it in one place? Or is it potentially going to be reused in other events?

**Review by WoLewicki — CHANGES_REQUESTED** [review]
Looks good! I left some comments, please answer them.

File: apple/Elements/RNSVGImage.mm (hunk 3 above)
Comment (bohdanprog) [reply apple/Elements/RNSVGImage.mm:224]: without that unfortunately the app crashed in old architecture.

File: apple/Elements/RNSVGImage.mm (hunk 2 above)
Comment (bohdanprog) [reply apple/Elements/RNSVGImage.mm:148]: okay, the good point :)

File: apple/Elements/RNSVGImage.mm (hunk 3 above)
Comment (WoLewicki) [reply apple/Elements/RNSVGImage.mm:224]: I meant `onLoadParamsForSource` method.

File: apple/Elements/RNSVGImage.mm (hunk 1 above)
Comment (WoLewicki) [reply apple/Elements/RNSVGImage.mm:149]: Ok let's do it like that then.

File: apple/Elements/RNSVGImage.mm (hunk 3 above)
Comment (bohdanprog) [reply apple/Elements/RNSVGImage.mm:224]: without that check, the app expects us to pass onLoad prop every time.

File: apple/Elements/RNSVGImage.mm (hunk 2 above)
Comment (bohdanprog) [reply apple/Elements/RNSVGImage.mm:139]: done

**Review by WoLewicki — APPROVED** [review]
So if it is now aligned between platforms and architectures, I think we are good to go :tada:
